### DISCUSSION TITLE
Add and use custom SASL plugin library

### DISCUSSION THREAD

**Discussion by dejan-kosak** [provectus/kafka-ui#1728] [Q&A] [answered]
I was wondering, is there a way to include our custom SASL plugin jar, which would later be used for configuring Kafka consumer?

```
security.protocol=SASL_PLAINTEXT
sasl.mechanism=OAUTHBEARER
// Configure class from custom SASL plugin library
sasl.login.callback.handler.class=com.myplugin.MyLoginCallback
```

If possible, could you give me more information how it is done.

Thank you!

**Comment by Haarolean** [comment, marked as answer]
Hi, you'll have to add a jar file with your classes to the classpath (either via editing the dockerfile and building the image yourself) or via runtime.

**Reply by robodenitro** [reply]
unfortunately `-jar` option can't be used with `-cp`, so we had to repack whole `kafka-ui-api` jar with our libraries

**Reply by Haarolean** [reply]
We're going to support plugging in jars within (#1816)

**Comment by sawou** [comment]
Hello @dejan-kosak 

did you solve this?
